src/plugins/vis_type_timeseries/public/application/components/data_format_picker.tsx

```diff
@@ -0,0 +1,311 @@
+              onChange={handleDurationChange('from')}
+              singleSelection={{ asPlainText: true }}
+              data-test-subj="dataFormatPickerDurationFrom"
+            />
+          </EuiFormRow>
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <EuiFormRow
+            id={htmlId('to')}
+            label={i18n.translate('visTypeTimeseries.dataFormatPicker.toLabel', {
+              defaultMessage: 'To',
+            })}
+          >
+            <EuiComboBox
+              isClearable={false}
+              options={durationOutputOptions}
+              selectedOptions={selectedTo ? [selectedTo] : []}
+              onChange={handleDurationChange('to')}
+              singleSelection={{ asPlainText: true }}
+              data-test-subj="dataFormatPickerDurationTo"
+            />
+          </EuiFormRow>
+        </EuiFlexItem>
+
+        {selectedTo?.value !== 'humanize' && (
+          <EuiFlexItem grow={false}>
+            <EuiFormRow
+              id={htmlId('decimal')}
+              label={i18n.translate('visTypeTimeseries.dataFormatPicker.decimalPlacesLabel', {
+                defaultMessage: 'Decimal places',
+              })}
+            >
+              <EuiFieldText
+                defaultValue={decimals}
+                placeholder={DEFAULT_OUTPUT_PRECISION}
+                onChange={handleDecimalsChange}
+                data-test-subj="dataFormatPickerDurationDecimal"
+              />
+            </EuiFormRow>
+          </EuiFlexItem>
+        )}
+      </>
+    );
+  }
+
+  let custom;
+  if (selectedFormatter === DATA_FORMATTERS.CUSTOM && shouldIncludeNumberOptions) {
+    custom = (
+      <EuiFlexItem grow={false}>
+        <EuiFormRow
+          label={
+            <FormattedMessage
+              id="visTypeTimeseries.dataFormatPicker.formatPatternLabel"
+              defaultMessage="Numeral.js format pattern (Default: {defaultPattern})"
+              values={{ defaultPattern: <EuiCode>{DEFAULT_CUSTOM_FORMAT_PATTERN}</EuiCode> }}
+            />
+          }
+          helpText={
+            <span>
+              <EuiLink target="_blank" href="http://numeraljs.com/#format">
+                <FormattedMessage
+                  id="visTypeTimeseries.dataFormatPicker.formatPatternHelpText"
+                  defaultMessage="Documentation"
+                />
+                &nbsp;
+                <EuiIcon type="link" />
+              </EuiLink>
+            </span>
+          }
+        >
+          <EuiFieldText
+            placeholder={DEFAULT_CUSTOM_FORMAT_PATTERN}
+            value={customFormatPattern}
+            onChange={handleCustomFormatStringChange}
+          />
+        </EuiFormRow>
+      </EuiFlexItem>
+    );
+  }
+
+  return (
+    <>
+      <EuiFlexItem>
+        <EuiFormRow
+          label={i18n.translate('visTypeTimeseries.defaultDataFormatterLabel', {
+            defaultMessage: 'Data Formatter',
```

Comment:
This should be sentence case.

```suggestion
            defaultMessage: 'Data formatter',
```